### DISCUSSION TITLE
TestClient Graphql MultiPart Upload

### DISCUSSION THREAD

**Discussion by nmatison** [fastapi/fastapi#3740] [Questions] [answered]
Hey everyone,

I'm struggling with getting the fast API test client to successfully utilize the aridane `Upload!` scalar provided via graphql-upload. Has anyone tried to do this before? I have code similar to the below: 

```python
def test_on_success(self, client: TestClient) -> None:
        with open(f"my_path_here", 'rb') as output:
            upload_json = {
                "query": """
                mutation($file: Upload! $type: DocumentType!) {
                  uploadDocument(type: $type, document: $file) {
                    id
                  }
                }
                """,
                "variables": {
                    "type": DocumentType.PAYMENT.value,
                    "file": None'
                },
            }
            upload_response = client.post("/graphql", data=upload_json, files={'file': ('payment.jpeg', output, 'image/jpeg', {'Expires': '0'})})
            breakpoint()
```

I keep getting `"Request 'operations' multipart field is not a valid JSON"` as a response no matter how much I play around with that `files` and `data` kwargs and what I'm feeding into them. 

I also tried something similar to https://github.com/tiangolo/fastapi/issues/1772 as well as quite a few other approaches from google, stackoverflow, etc. 

Any help or direction would be greatly appreciated!

**Comment by YuriiMotov** [comment, marked as answer]
I think converting `variables` field of `upload_json` should help.
Then you need to parse it manually (convert to `dict`) on backend side
